rust/lance-index/src/vector/ivf.rs

```diff
@@ -306,12 +306,16 @@ impl<T: ArrowFloatType + Dot + L2 + Cosine + 'static> IvfImpl<T> {
 
         let dimension = data.ndim();
         let centroids = self.centroids.data();
-        let data = data.data();
-        let metric_type = self.metric_type;
-
         let num_centroids = centroids.len() / dimension;
-        let num_rows = data.len() / dimension;
+        let num_rows = data.num_rows();
+
+        let (data, metric_type) = if self.metric_type == MetricType::Cosine {
```

Comment:
should we move this higher? maybe to `search`?

Reply:
ok i can try

Reply:
THe shuffler is still using this piece separately, tho